curves/bls12_377/src/curves/mod.rs

```diff
@@ -32,6 +30,44 @@ pub trait HostFunctions: 'static {
     fn bls12_377_mul_affine_g2(base: Vec<u8>, scalar: Vec<u8>) -> Vec<u8>;
 }
 
+pub struct Config<H: HostFunctions>(PhantomData<H>);
+
+pub type Bls12_377<H> = Bls12<Config<H>>;
+
+fn type_name_of<T>(_: &T) -> &str {
+    core::any::type_name::<T>()
+}
+
+impl<H: HostFunctions> Config<H> {
+    // Hack to serialize all the expected types into affine.
```

Comment:
This is the hack I was talking about in the PR description.

Note that before the conversion we check the type... so it should be a fairly safe operation.

TODO: is there a better way to check the types?

Reply:
If we absolutely have to do this then maybe it'd be nice to add a helper for this. And if possible use `TypeId::of`, which is safer. (`type_name` ignores lifetimes and it's *technically* not guaranteed that two different types will have unique names).

```rust
// Not entirely safe wrapper.
unsafe fn safeish_transmute<T, U>(input: T) -> Result<U, T> {
    if core::any::type_name::<T>() == core::any::type_name::<U>() && 
       core::mem::size_of::<T>() == core::mem::size_of::<U>() {
        let output = core::mem::transmute_copy(&input);
        core::mem::forget(input);
        Ok(output)
    } else {
        Err(input)
    }
}

// This should be always safe, I think. But it requires 'static bounds.
fn safe_transmute<T, U>(input: T) -> Result<U, T> where T: 'static, U: 'static {
    if core::any::TypeId::of::<T>() == core::any::TypeId::of::<U>() {
        let output = unsafe {
            core::mem::transmute_copy(&input)
        };
        core::mem::forget(input);
        Ok(output)
    } else {
        Err(input)
    }
}
```

Reply:
Unfortunately I don't think we can bound T to be `'static` since the type is given by the trait method we are implementing (i.e. `multi_miller_loop`)

Reply:
Indeed I've tried to use `TypeId` but looks like it requires T to be `'static`